torch/distributed/checkpoint/filesystem.py

```diff
@@ -108,7 +108,7 @@ def values(self) -> Iterator[tuple[torch.Tensor, object]]:
         for _, obj in self.items:
             tensor = self.resolve_fun(obj).detach()
             tensor = tensor.cpu()
-            if tensor.storage().size() != tensor.numel():
+            if tensor.untyped_storage().size() != tensor.numel()* tensor.element_size():
                 tensor = tensor.clone()
```

Comment:
Huh, what is this clone support to accomplish? Could specify it as part of the optional cpu() if needed above.
```suggestion
            tensor = tensor.to(device="cpu", copy=tensor.untyped_storage().size() != tensor.numel()* tensor.element_size())
```